### ISSUE TITLE
alpine setup not completing and extensions issue

### ISSUE THREAD

**Issue by seano-vs** [feelfreelinux/octo4a#385]
TL;DR:

It seems that I have a combination of #83 and #211.

I am able to install the apk (through adb as well as directly in android) and it seems to run. When I install octoprint, it does the "jump to 5%0 and then close" issue where it cites a corrupt installation on re-launch. It should be noted that all of this happens prior to alpine running, so there is no SSH. Here is the log from trying to run the installation:

```sh
03-05 07:12:31.070  9030  9102 D 💻    : No bootstrap detected, proceeding with installation
03-05 07:12:31.633  9030  9123 D 💻    : Downloading bootstrap 0.2.1
03-05 07:12:32.547  9030  9123 D 💻    : Bootstrap extracted, setting it up...
03-05 07:12:32.590  9030  9123 D 🐸    : add-user.sh
03-05 07:12:32.590  9030  9123 D 🐸    : fake_proc_stat
03-05 07:12:32.591  9030  9123 D 🐸    : install-bootstrap.sh
03-05 07:12:32.591  9030  9123 D 🐸    : ioctlHook.so
03-05 07:12:32.591  9030  9123 D 🐸    : root
03-05 07:12:32.591  9030  9123 D 🐸    : root-pre5
03-05 07:12:32.592  9030  9123 D 🐸    : rootfs.tar.xz
03-05 07:12:32.592  9030  9123 D 🐸    : run-bootstrap.sh
03-05 07:12:32.984  9030  9123 D 🐸    : bootstrap ready, run with run-bootstrap.sh
03-05 07:12:33.133  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.160  9030  9123 D 🐸    : Welcome to Alpine!
03-05 07:12:33.161  9030  9123 D 🐸    : 
03-05 07:12:33.164  9030  9123 D 🐸    : The Alpine Wiki contains a large amount of how-to guides and general
03-05 07:12:33.166  9030  9123 D 🐸    : information about administrating Alpine systems.
03-05 07:12:33.166  9030  9123 D 🐸    : See <https://wiki.alpinelinux.org/>.
03-05 07:12:33.167  9030  9123 D 🐸    : 
03-05 07:12:33.168  9030  9123 D 🐸    : You can setup the system with the command: setup-alpine
03-05 07:12:33.169  9030  9123 D 🐸    : 
03-05 07:12:33.170  9030  9123 D 🐸    : You may change this message by editing /etc/motd.
03-05 07:12:33.171  9030  9123 D 🐸    : 
03-05 07:12:33.256  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.275  9030  9123 D 🐸    : /bin/sh: apk: not found
03-05 07:12:33.358  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.379  9030  9123 D 🐸    : /bin/sh: can't create /etc/ssh/sshd_config: nonexistent directory
03-05 07:12:33.467  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.503  9030  9123 D 🐸    : /bin/sh: ssh-keygen: not found
03-05 07:12:33.506  9030  9123 D 💻    : Bootstrap installation done
03-05 07:12:33.508  9030  9123 D 💻    : Bootstrap installed
03-05 07:12:33.615  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.644  9030  9123 D 🐸    : /bin/sh: curl: not found
03-05 07:12:33.648  9030  9123 D 🐸    : /bin/sh: bash: not found
03-05 07:12:33.759  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.785  9030  9123 D 🐸    : /bin/sh: curl: not found
03-05 07:12:33.873  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:33.900  9030  9123 D 🐸    : unzip: can't open octoprint.zip[.zip]
03-05 07:12:34.044  9030  9123 D 🐸    : proot warning: can't sanitize binding "/data/data/com.octo4a/files/serialpipe": No such file or directory
03-05 07:12:34.071  9030  9123 D 🐸    : /bin/sh: cd: line 0: can't cd to Octo*: No such file or directory
```

I've also noticed how the extensions tab fails as well, causing the app to close. Here is the output from that:

```sh
03-05 07:17:12.251  9209  9209 D AndroidRuntime: Shutting down VM
03-05 07:17:12.252  9209  9209 E AndroidRuntime: FATAL EXCEPTION: main
03-05 07:17:12.252  9209  9209 E AndroidRuntime: Process: com.octo4a, PID: 9209
03-05 07:17:12.252  9209  9209 E AndroidRuntime: java.lang.NullPointerException: Attempt to get length of null array
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at com.octo4a.repository.ExtensionsRepositoryImpl.getValidExtensions(ExtensionsRepository.kt:213)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at com.octo4a.ui.fragments.ExtensionsFragment.onViewCreated(ExtensionsFragment.kt:32)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2974)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:543)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2177)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2094)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1990)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7854)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
03-05 07:17:12.252  9209  9209 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183)
03-05 07:17:12.310   443  2406 W ActivityTaskManager:   Force finishing activity com.octo4a/.ui.MainActivity
03-05 07:17:12.310   443  9323 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
03-05 07:17:12.314   251   251 I gralloc : alloc buffer:W:1920 H:1080 S:1920 PL[1920 1080] SZ:8294400 fmt:1 if:0 af:1 usg:b66 fg:00000004
03-05 07:17:12.329   443   506 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
03-05 07:17:12.330  9209  9209 I Process : Sending signal. PID: 9209 SIG: 9
03-05 07:17:12.432   443  5554 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(uid/pid:10097/9209, android.os.BinderProxy@175d0b1)
03-05 07:17:12.433   443   589 D ConnectivityService: releasing NetworkRequest [ REQUEST id=121, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VCN_MANAGED Uid: 10097 RequestorUid: 10097 RequestorPkg: com.octo4a] ] (release request)
03-05 07:17:12.433   443  5712 I WindowManager: WIN DEATH: Window{d8e299a u0 com.octo4a/com.octo4a.ui.MainActivity}
03-05 07:17:12.433   443   724 I ActivityManager: Process com.octo4a (pid 9209) has died: fg  TOP 
03-05 07:17:12.433   443  5712 W InputManager-JNI: Input channel object 'd8e299a com.octo4a/com.octo4a.ui.MainActivity (client)' was disposed without first being removed with the input manager!
01-01 10:47:34.806   130   130 W binder_cleanup_transaction: 12 callbacks suppressed
01-01 10:47:34.806   130   130 I binder  : undelivered transaction 1605954, process died.
01-01 10:47:34.806   130   130 I binder  : undelivered transaction 1605973, process died.
03-05 07:17:12.439   240   240 I Zygote  : Process 9209 exited due to signal 9 (Killed)
03-05 07:17:12.439   443   508 I libprocessgroup: Successfully killed process cgroup uid 10097 pid 9209 in 0ms
03-05 07:17:12.444   443   724 W ActivityManager: Scheduling restart of crashed service com.octo4a/.service.OctoPrintService in 1000ms for start-requested
03-05 07:17:12.448   266   266 E BpTransactionCompletedListener: Failed to transact (-32)
03-05 07:17:12.457   659  9329 W RingtonePlayer: STOP command without a player
03-05 07:17:12.468   443   724 W ActivityTaskManager: Can't find TaskDisplayArea to determine support for multi window. Task id=289 attached=false
03-05 07:17:12.468   443   724 W ActivityTaskManager: Can't find TaskDisplayArea to determine support for multi window. Task id=289 attached=false
03-05 07:17:12.490   251   251 I gralloc : alloc buffer:W:1920 H:1080 S:1920 PL[1920 1080] SZ:8294400 fmt:1 if:0 af:1 usg:20000b00 fg:00000004
03-05 07:17:12.491   251   251 I gralloc : alloc buffer:W:1920 H:1080 S:1920 PL[1920 1080] SZ:8294400 fmt:1 if:0 af:1 usg:20000b00 fg:00000004
03-05 07:17:12.491   443   495 W ActivityManager: setHasOverlayUi called on unknown pid: 9209
03-05 07:17:12.493   251   251 I gralloc : alloc buffer:W:1920 H:1080 S:1920 PL[1920 1080] SZ:8294400 fmt:1 if:0 af:1 usg:20000b00 fg:00000004
03-05 07:17:12.515   443   494 W IorapForwardingService: No service published for: iorapd
03-05 07:17:12.813   443   495 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{13d71ea u0 com.octo4a/.ui.MainActivity t-1 f}}
03-05 07:17:12.902   443   465 I system_server: NativeAlloc concurrent copying GC freed 259159(11MB) AllocSpace objects, 23(668KB) LOS objects, 50% free, 15MB/31MB, paused 238us total 486.839ms
03-05 07:17:12.914   443   467 W System  : A resource failed to call release. 
03-05 07:17:12.916   443   467 W System  : A resource failed to call close. 
03-05 07:17:12.916   443   467 W System  : A resource failed to call release. 
03-05 07:17:13.026   443   443 D HdmiCecLocalDevice: AWAW onMessage message: <Give Device Power Status> 04:8F
03-05 07:17:13.135   258   258 D hdmi-cec: param.msg: 40-90-2-0-0, param.tx_status = 1
03-05 07:17:13.448   443   506 D CompatibilityChangeReporter: Compat change id reported: 135634846; UID 10097; state: DISABLED
03-05 07:17:13.449   443   507 D CompatibilityChangeReporter: Compat change id reported: 143937733; UID 10097; state: DISABLED
03-05 07:17:13.462   240   240 D Zygote  : Forked child process 9332
03-05 07:17:13.470   443   507 I ActivityManager: app.processName = com.octo4a
03-05 07:17:13.470   443   507 I ActivityManager: Start proc 9332:com.octo4a/u0a97 for service {com.octo4a/com.octo4a.service.OctoPrintService}
03-05 07:17:13.471  9332  9332 I Zygote  : seccomp disabled by setenforce 0
03-05 07:17:13.500   352   364 I adbd    : jdwp connection from 9332
03-05 07:17:13.520   443   494 W IorapForwardingService: No service published for: iorapd
03-05 07:17:13.525  9332  9332 D CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10097; state: DISABLED
03-05 07:17:13.522   258   258 I EventThread: type=1400 audit(0.0:2420): avc: denied { search } for name="extcon" dev="sysfs" ino=18073 scontext=u:r:hal_tv_cec_default:s0 tcontext=u:object_r:sysfs_extcon:s0 tclass=dir permissive=1
03-05 07:17:13.538  9332  9332 D ApplicationLoaders: Returning zygote-cached class loader: /system/framework/android.test.base.jar
03-05 07:17:13.561  9332  9332 V GraphicsEnvironment: ANGLE Developer option for 'com.octo4a' set to: 'default'
03-05 07:17:13.561  9332  9332 V GraphicsEnvironment: Neither updatable production driver nor prerelease driver is supported.
03-05 07:17:13.569  9332  9332 D NetworkSecurityConfig: No Network Security Config specified, using platform default
03-05 07:17:13.570  9332  9332 D NetworkSecurityConfig: No Network Security Config specified, using platform default
03-05 07:17:13.573  9332  9332 I MultiDex: VM with version 2.1.0 has multidex support
03-05 07:17:13.573  9332  9332 I MultiDex: Installing application
03-05 07:17:13.573  9332  9332 I MultiDex: VM has multidex support, MultiDex support library is disabled.
03-05 07:17:13.583  9332  9332 I [Koin]  : [init] declare Android Context
03-05 07:17:13.597  9332  9332 I [Koin]  : loaded 14 definitions - 0.718375 ms
03-05 07:17:13.602   443   443 I Thread-25: type=1400 audit(0.0:2421): avc: denied { call } for scontext=u:r:system_server:s0 tcontext=u:r:hal_awdisplayoutput_default:s0 tclass=binder permissive=1
03-05 07:17:13.670  9353  9353 I which   : type=1400 audit(0.0:2422): avc: denied { getattr } for path="/system/xbin/su" dev="dm-0" ino=3496 scontext=u:r:untrusted_app_27:s0:c97,c256,c512,c768 tcontext=u:object_r:su_exec:s0 tclass=file permissive=1 app=com.octo4a
03-05 07:17:13.670  9353  9353 I which   : type=1400 audit(0.0:2423): avc: denied { execute } for name="su" dev="dm-0" ino=3496 scontext=u:r:untrusted_app_27:s0:c97,c256,c512,c768 tcontext=u:object_r:su_exec:s0 tclass=file permissive=1 app=com.octo4a
```

I hope these issues are related, and will start digging into them more

Here are the specs of this device:

Android 12.1.A5

Kernel version 5.4.125

**Comment by alufers** [comment]
This should be fiexed in 1.2.3.
